sympy/ntheory/factor_.py

```diff
@@ -1413,6 +1413,8 @@ def eval(cls, n):
             for p, k in factors.items():
                 t *= (p - 1) * p**(k - 1)
             return t
+        elif not n.is_Symbol:
+            raise ValueError("n must be a positive integer")
```

Comment:
I am not sure why have you used `is_Symbol` for testing non-integer.
As it fails in case of generic symbols
```python
In [7]: x = Symbol('x', integer=False)
In [8]: totient(x)
Out[8]: totient(x)
```